.travis.yml

```diff
@@ -35,6 +36,14 @@ script:
       pdflatex std;
     fi
   - popd
+  # Fail if there is whitespace at the ends of any lines
+  - export TEST_STATUS=0;
+    if [ "$BUILD_TYPE" = "check-whitespace" ]; then
+      grep '\s$' source/*.tex && export TEST_STATUS=1;
```

Comment:
@tkoeppe When I tried that previously it didn't work, but it appears to be working now. (I guess it was a combination of bugs previously—testing the `.travis-ci.yml` file is a bit of trial-and-error.) Thanks for prompting me to give it another shot; it greatly simplifies the code!